synapse/storage/databases/main/room.py

```diff
@@ -1388,6 +1464,9 @@ async def maybe_store_room_on_outlier_membership(
         # mark the room as having an auth chain cover index.
         has_auth_chain_index = await self.has_auth_chain_index(room_id)
 
+        create_event = await self.get_create_event_for_room(room_id)
+        room_creator = create_event.content.get("creator", None)
```

Comment:
I'm not so sure about this case being totally right. The MSC2716 stuff (Complement test suite) doesn't touch this code at all to be sure. And there are many usages of this function to check out.

For example with the `upsert_room_on_join` case above, I had to change it to pass in the `auth_events` because no state events were persisted at that time to use `get_create_event_for_room` which derives from them. But `upsert_room_on_join` also only has one usage so it's easy to reason about and refactor.

Reply:
Moved to https://github.com/matrix-org/synapse/pull/10697